autoload/pythonimports.vim

```diff
@@ -6,12 +6,35 @@ function! pythonimports#filename2module(filename)
   " directly included in PYTHONPATH.
   let pkg = fnamemodify(a:filename, ":p")
   let root = fnamemodify(pkg, ":h")
-  while strlen(root)
-    if !filereadable(root . "/__init__.py")
-      break
-    endif
-    let root = fnamemodify(root, ":h")
-  endwhile
+
+  let found = 0
+  if exists("g:python_paths")
+      for path in g:python_paths
+            if root =~ path
+                "path detected
+                while strlen(root)
+                    if root == path
+                        "echo 'found '. root
+                        let found = 1
+                        break
+                    endif
+                    let root = fnamemodify(root, ":h")
```

Comment:
You're modifying the same local variable here.  Eventually it becomes an empty string.  Then, on the next iteration of the `for` loop, you'll be checking the empty string instead of the original root.  That seems like a bug?

Reply:
This is the bug I want fixed before merging.

Reply:
right, need to fix it. I'll make a commit.